versioned_docs/version-V3/guides/providing-liquidity/setting-up-your-contract.md

```diff
@@ -0,0 +1,188 @@
+---
+id: setting-up
+title: Setting Up Your Contract
+sidebar_position: 1
+---
+
+## Setting up the Contract
+
+First we declare the solidity version that will be used to compile the contract, and the `abicoder v2` to allow arbitrary nested arrays and structs to be encoded and decoded in calldata, a feature we use when transacting with a pool.
+
+```solidity
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+pragma abicoder v2;
+```
+
+Next, import the two contracts we'll be using from the npm package installation
+
+```solidity
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+import '@uniswap/v3-core/contracts/libraries/TickMath.sol';
+import '@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol';
+import '@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol';
+import '@uniswap/v3-periphery/contracts/interfaces/INonfungiblePositionManager.sol';
+import '@uniswap/v3-periphery/contracts/libraries/TransferHelper.sol';
+import '@uniswap/v3-periphery/contracts//base/LiquidityManagement.sol';
+```
+
+Here we create our contract called `LiquidityExamples`, and inherit both `IERC721Receiver` and `LiquidityManagement`.
+
+Here we hardcode the token contract addresses and pool fee tiers for our example. In production, you would likely use in input parameter for this and pass the input into a memory variable, allowing you to change what the pools and tokens you are interacting with on a per transaction basis, but for conceptual simplicity we are hardcoding them here.
+
+```solidity
+contract LiquidityExamples is IERC721Receiver, LiquidityManagement {
```

Comment:
quick deletion of LiquidityManagement (we no longer call the function from that contract)